vertical-pod-autoscaler/pkg/recommender/routines/recommender.go

```diff
@@ -108,25 +108,18 @@ func (r *recommender) UpdateVPAs() {
 		vpa.UpdateConditions(hasMatchingPods)
 		if err := r.clusterState.RecordRecommendation(vpa, time.Now()); err != nil {
 			klog.Warningf("%v", err)
-			if klog.V(4).Enabled() {
-				klog.Infof("VPA dump")
-				klog.Infof("%+v", vpa)
-				klog.Infof("HasMatchingPods: %v", hasMatchingPods)
-				klog.Infof("PodCount: %v", vpa.PodCount)
-				pods := r.clusterState.GetMatchingPods(vpa)
-				klog.Infof("MatchingPods: %+v", pods)
-				if len(pods) != vpa.PodCount {
-					klog.Errorf("ClusterState pod count and matching pods disagree for VPA %s", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName))
-				}
+			pods := r.clusterState.GetMatchingPods(vpa)
+			klog.InfoS("VPA dump", "vpa", vpa, "hasMatchingPods", hasMatchingPods, "PodCount", vpa.PodCount, "MatchingPods", pods)
```

Comment:
How about we use this opportunity to unify the casing of the log fields. So making this `podCount` and `matchingPods`?

Reply:
👍